adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -301,17 +302,30 @@ final void handleEndpointConnection(final MqttEndpoint endpoint) {
         }
     }
 
-    private void handleConnectionRequestResult ( final MqttEndpoint endpoint, final AsyncResult<Device> done) {
+    private void handleConnectionRequestResult(final MqttEndpoint endpoint,
+            final AsyncResult<Device> authenticationResult) {
 
-        if ( done.succeeded() ) {
+        if (authenticationResult.succeeded()) {
 
-            endpoint.accept(false);
-            sendConnectedEvent(endpoint.clientIdentifier(), done.result());
+            sendConnectedEvent(endpoint.clientIdentifier(), authenticationResult.result())
+                    .setHandler(result -> {
+                        if (result.succeeded()) {
+                            endpoint.accept(false);
+                        } else {
+                            LOG.info(
+                                    "connection request from client [clientId: {}] rejected due to connection event "
+                                            + "failure: {}",
+                                    endpoint.clientIdentifier(),
+                                    MqttConnectReturnCode.CONNECTION_REFUSED_SERVER_UNAVAILABLE,
+                                    authenticationResult.cause());
```

Comment:
IMHO this needs to be `result.cause()` to make sense.

In order to improve readability, I also propose to rename `authenticationResult` to `authenticationAttempt` and `result` to `sendAttempt`.